ERCS/erc-7715.md

```diff
@@ -0,0 +1,601 @@
+eg. a token allowance, a gas limit, an access list, an ENS record, etc.
+
+For each permission there will be a set of policies and for each policy there will be a set of properties which are specified as follows:
+
+```typescript
+interface Permission {
+  type: name;
+  policies: Policy[];
+  required: boolean;
+  data: any;
+}
+
+interface Policy {
+  type: name;
+  data: any;
+}
+```
+
+#### Request Specification
+
+```tsx
+type GrantPermissionsRequestParams = {
+  account?: `0x${string}`;
+
+  chainId: number;
+
+  signer: {
+    type: string;
+    data: any;
+  };
+
+  permissions: Permission[];
+
+  expiry: number;
+};
+```
+
+`account` is an optional field that identifies the account being targetted for this permissions request which is useful when a connection has been established and multiple accounts have been exposed. 
+
+`chainId` is field that identifies that chain with EIP-155 which these permissions are contextual and all addressed can be found for the accounts and permissions.
+
+`signer` is a field that identifies the key or account associated with the session or alternatively the wallet will manage the session. See the “Wallet-managed Sessions” section for details.
+
+`permissions` is a list of permissions with determined policies. See “Policy Types” section for details.
+
+Each permission can be either `required` or not. If a permission is `required`, the wallet MUST grant the permission if it approves the request. Otherwise, the wallet may reject the permission even if it approves the request (presumably because it grants other permissions).
+
+`expiry` is a UNIX timestamp (in seconds) that specifies the time by which this session MUST expire.
+
+Request example:
+
+```tsx
+{
+  account: '0x...'
+
+  chainId: 123,
+
+  signer: {
+    type: 'key',
+    data: {
+      id:'did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme',
+    }
+  },
+
+  permissions: [
+    {
+      type: 'native-token-transfer',
+      policies: [
+        {
+          type: 'gas-limit',
+          data: {
+            limit: '0x...',
+          },
+        },
+        {
+          type: 'call-limit',
+          data: {
+            count: 1,
+          },
+        },
+      ],
+      required: true,
+      data: {
+        address: '0x...',
+        allowance: '500000000'
+      }
+    },
+    {
+      type: 'erc20-token-transfer',
+      policies: [
+        {
+          type: 'gas-limit',
+          data: {
+            limit: '0x...',
+          },
+        }
+      ],
+      required: false,
+      data: {
+        address: '0x...',
+        allowance: '2000000000'
+      }
+    }
+  ],
+
+  expiry: 1577840461
+}
+```
+
+#### Response Specification
+
+```tsx
+type GrantPermissionsResponse = {
+  grantedPermissions: Permission[];
+
+  expiry: number;
+
+  signerData: {
+    userOpBuilder?: `0x${string}`;
+    managerAddress?: `0x${string}`;
+  };
+
+  factory?: `0x${string}`;
+
+  factoryData?: `0x${string}`;
```

Comment:
Can we bundle these together within a new optional field `accountData`? I don't see why these should be optional individually and it would be consistent with the pattern established with `signerData`.
```suggestion
  accountData?: {
    factory?: `0x${string}`;
    factoryData?: `0x${string}`;
  }
```
```suggestion
  factory?: `0x${string}`;

  factoryData?: `0x${string}`;
```

Reply:
or maybe we just call both `signerMeta` and `accountMeta`

Reply:
`signerMeta` should also be optional for `wallet` signer type